core/comp_ui.py

```diff
@@ -368,13 +375,13 @@ def _ReturnToPrompt(self, num_lines):
     self.f.flush()
 
   def _PrintCandidates(self, unused_subst, matches, unused_max_match_len):
-    # type: (Optional[Any], List[str], Optional[Any]) -> None
+    # type: (Optional[str], List[str], int) -> None
     term_width = self._GetTerminalWidth()
 
     # Variables set by the completion generator.  They should always exist,
     # because we can't get "matches" without calling that function.
     display_pos = self.comp_state.display_pos
-    self.debug_f.log('DISPLAY POS in _PrintCandidates = %d', display_pos)
+    self.debug_f.write('DISPLAY POS in _PrintCandidates = %d' % display_pos)
```

Comment:
I think this needs a `\n` to match log()

I guess we should get rid of the method `DebugFile::log()` in favor of `%` ?   

I have been meaning to do that for some other instances

varargs in different functions are annoying to translate -- I think we could be consistent and use `%` everywhere

Reply:
ah yeah. i have a plan to convert the other `log()`s to `write()`s as i translate other files. then i'll remove `DebugFile::log()` once all of the call sites have been translated